### PR TITLE
Add upgrade instructions from v3 to v4

### PR DESCRIPTION
**Pull request by ccremer**
relevant upgrade instructions from #14 

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/upgrade-3.x-to-4.x.adoc [hunk 1]
```diff
@@ -0,0 +1,90 @@
+= Upgrade from 3.x to 4.x
+
+This guide describes the steps to perform an upgrade of the component from version 3.x to 4.x.
+
+The major changes affect the ingress feature and configuration.
+This component now features re-encrypted ingress by default resp. encrypts cluster-internal traffic.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/upgrade-3.x-to-4.x.adoc:6]:
```suggestion
This component now features re-encrypted ingress by default.
That is, traffic between the ingress controller and Keycloak is now encrypted by default.
```

File: docs/modules/ROOT/pages/how-tos/upgrade-3.x-to-4.x.adoc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+The sections below explain more.
+
+=== Changed parameters
+
+* `ingress.secretName` has moved to `ingress.tls.secretName`.
+* `ingress.enabled` changed to `true` by default.
+* `ingress.annotations` received several annotations depending on `tls.provider` and `tls.termination`.
+* `helm_values.extraVolumes` and `helm_values.extraVolumeMounts` got an additional volume/mount from the `tls.secretName` secret.
+* `helm_values.ingress.servicePort` changed to `https` by default.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/upgrade-3.x-to-4.x.adoc:25]: Might be worth mentioning more explicitly that users must update their configuration to match these new parameters if they've already configured some of them.

File: docs/modules/ROOT/pages/how-tos/upgrade-3.x-to-4.x.adoc [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      provider: vault
+----
+
+. Generate new long-running self-signed certificate and xref:how-tos/keycloak-tls.adoc[store in Vault].
+
+=== Pass-through TLS with Let's Encrypt
+
+> User traffic > Ingress controller (pass-through) > Keycloak (with LE certificate)
+
+NOTE: This mode requires an ingress controller that supports SSL/TLS pass-through.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/upgrade-3.x-to-4.x.adoc:55]: Maybe give an example on how to enable pass-through for component-ingress-nginx (IIRC there is one in the parameters documentation already).

**Review by simu — APPROVED** [review]
LGTM

**Review by tobru — APPROVED** [review]

**Pull request merged**